DNN Platform/Library/Common/Utilities/FileExtensionWhitelist.cs

```diff
@@ -141,5 +141,19 @@ private IEnumerable<string> NormalizeExtensions(IEnumerable<string> additionalEx
         {
             return additionalExtensions.Select(ext => (ext.StartsWith(".") ? ext : "." + ext).ToLowerInvariant());
         }
+
+        public FileExtensionWhitelist RestrictBy(FileExtensionWhitelist parentList)
+        {
+            var filter = parentList.AllowedExtensions.ToList();
+            var filteredList = new List<string>();
+            foreach (var ext in _extensions)
+            {
+                if (filter.Contains(ext))
+                {
+                    filteredList.Add(ext.Substring(1));
+                }
+            }
+            return new FileExtensionWhitelist(string.Join(",", filteredList));
```

Comment:
If this gets called a lot, we may want to switch to a `StringBuilder` (or, at least, do a little performance testing)

Reply:
This could also possibly be done more succintly with something like
var allowedExtensions = _extensions.Where(x => filter.Contains(x)).Select(s => s.Substring(1));

Just could be abit more condensed.readable.